libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -47,76 +48,215 @@ typedef struct MqttPublishInfo MQTTPublishInfo_t;
 struct MQTTPacketInfo;
 typedef struct MQTTPacketInfo MQTTPacketInfo_t;
 
+/**
+ * @brief Signature of the transport interface receive function.
+ *
+ * A function with this signature must be provided to the MQTT library to read
+ * data off the network.
+ *
+ * @param[in] context The context provided with this function.
+ * @param[out] pBuffer Buffer to receive network data.
+ * @param[in] bytesToRecv Bytes to receive from the network. pBuffer must be at
+ * least this size.
+ *
+ * @return The number of bytes received; negative value on failure.
+ */
 typedef int32_t (* MQTTTransportRecvFunc_t )( MQTTNetworkContext_t context,
                                               void * pBuffer,
                                               size_t bytesToRecv );
 
+/**
+ * @brief Return codes from MQTT functions.
+ */
 typedef enum MQTTStatus
 {
-    MQTTSuccess = 0,
-    MQTTBadParameter,
-    MQTTNoMemory,
-    MQTTSendFailed,
-    MQTTRecvFailed,
-    MQTTBadResponse,
-    MQTTServerRefused
+    MQTTSuccess = 0,    /**< Function completed successfully. */
+    MQTTBadParameter,   /**< At least one parameter was invalid. */
+    MQTTNoMemory,       /**< A provided buffer was too small. */
+    MQTTSendFailed,     /**< The transport send function failed. */
+    MQTTRecvFailed,     /**< The transport receive function failed. */
+    MQTTBadResponse,    /**< An invalid packet was received from the server. */
+    MQTTServerRefused   /**< The server refused a CONNECT or SUBSCRIBE. */
 } MQTTStatus_t;
 
+/**
+ * @brief MQTT Quality of Service values.
+ */
 typedef enum MQTTQoS
 {
-    MQTTQoS0 = 0,
-    MQTTQoS1 = 1,
-    MQTTQoS2 = 2
+    MQTTQoS0 = 0,    /**< Delivery at most once. */
+    MQTTQoS1 = 1,    /**< Delivery at least once. */
+    MQTTQoS2 = 2     /**< Delivery exactly once. */
 } MQTTQoS_t;
 
+/**
+ * @brief Buffer passed to MQTT library.
+ *
+ * These buffers are not copied and must remain in scope for the duration of the
+ * MQTT operation.
+ */
 struct MQTTFixedBuffer
 {
-    uint8_t * pBuffer;
-    size_t size;
+    uint8_t * pBuffer;    /**< @brief Pointer to buffer. */
+    size_t size;          /**< @brief Size of buffer. */
 };
 
+/**
+ * @brief MQTT CONNECT packet parameters.
+ */
 struct MQTTConnectInfo
 {
+    /**
+     * @brief Whether to establish a new, clean session or resume a previous session.
+     */
     bool cleanSession;
```

Comment:
Given the C89 guidelines, we probably cannot use bool. Not required in this PR though.